### DISCUSSION TITLE
Regression for multi form parameters

### DISCUSSION THREAD

**Discussion by tonnico** [fastapi/fastapi#12192] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import FastAPI, Form, Request

app = FastAPI()


@app.get("/")
def index(request: Request, multi: Annotated[list[str], Form()]): ...
```

### Description

Open Browser
Call: http://127.0.0.1:8080/?multi=foo
Response: `{"detail":[{"type":"missing","loc":["body","multi"],"msg":"Field required","input":null}]}`

### Operating System

macOS

### Operating System Details

Looks like a regression with 0.114.0.

Until 0.113.0 it was working as expected.

### FastAPI Version

0.114.1

### Pydantic Version

2.9.1

### Python Version

Python 3.12.5

**Comment by sinisaos** [comment]
@Wurstnase Sorry if I missed your point, but why do you use a `Form` in a `GET` request to get the query parameters (in my case it doesn't work in the latest or previous versions of FastAPI because it throws a validation error with status code 422) . Shouldn't you be using a `Query` to get query parameters like this
```python
from typing import Annotated
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/")
def index(multi: Annotated[list[str], Query()]):
    return multi
```
If you use a `Form` in a `POST` request like this
```python
from typing import Annotated
from fastapi import FastAPI, Form

app = FastAPI()


@app.post("/")
def index(multi: Annotated[list[str], Form()]):
    return multi
```

everything works.

**Reply by tonnico** [reply]
@sinisaos , many thanks for looking into it.

True, looks like you are right. I will check my minimal example. 

On our app we use POST with a router and multi where I see the same exception.
Accidentally I thought it may be the same with GET.

I will try to reproduce it in a second minimal example.

**Reply by sinisaos** [reply]
@Wurstnase No worries. In your example with `POST` request, everything works fine in my case. I see people reporting regressions in the latest versions, so maybe that's the case for you too. If you can make another MRE I will try.

**Reply by tonnico** [reply]
@sinisaos I got it finally.

It depends on the `alias`. It seems to stop working with 0.114.0.

```python
from typing import Annotated

import fastapi
from fastapi import FastAPI, Form
from fastapi import Request
from fastapi.testclient import TestClient


app = FastAPI()

@app.post("/ok")
def multi_ok(
    request: Request,
    multi: Annotated[list[str], Form(default_factory=list)],
):
    return multi

@app.post("/fail")
def multi_fail(
    request: Request,
    multi: Annotated[list[str], Form(alias="mul-ti", default_factory=list)],
):
    return multi

if __name__ == "__main__":
    client = TestClient(app)
    r = client.post("/ok", data={"multi": "1"})
    assert r.status_code == 200

    r = client.post("/fail", data={"mul-ti": "1"})
    if fastapi.__version__ < "0.114.0":
        assert r.status_code == 200
    else:
        assert r.status_code == 422
```

**Reply by tonnico** [reply]
I found some more details:

```python
    client = TestClient(app)
    r = client.post("/ok", data={"multi": "1"})
    assert r.status_code == 200
    assert r.json() == ["1"]

    r = client.post("/fail", data={"mul-ti": ["1"]})
    version = fastapi.__version__
    if version < "0.113.0":
        assert r.status_code == 200
        assert r.json() == ["1"]
    elif version < "0.114.0":
        assert r.status_code == 200
        assert r.json() == []
    else:  # >= 0.114.0
        assert r.status_code == 422
```

**Reply by tonnico** [reply, marked as answer]
> @Wurstnase No worries. In your example with `POST` request, everything works fine in my case. I see people reporting regressions in the latest versions, so maybe that's the case for you too. If you can make another MRE I will try.

@sinisaos I just found your solution in https://github.com/fastapi/fastapi/discussions/12139. That should be the same problem.

**Reply by sinisaos** [reply]
@Wurstnase I tried your example and yes, that change will solve your problem. As described in that discussion, I think `filed.alias` should be the correct choice because it satisfies both `field.name` and `field.alias`. The results are correct
```bash
> curl  --form multi=1,2,3 http://127.0.0.1:8000/ok
["1,2,3"]
> curl  --form mul-ti=1,2,3 http://127.0.0.1:8000/fail
["1,2,3"]
```
